app/models/component.rb

```diff
@@ -0,0 +1,25 @@
+# == Schema Information
+#
+# Table name: components
+#
+#  id            :integer          not null, primary key
+#  type          :integer
+#  audio_url     :string
+#  content_url   :string
+#  position      :integer
+#  subsection_id :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Component < ActiveRecord::Base
+  enum type: [ :slide, :form, :multimedia ]
+
+  validates :type, presence: true
+  validates :content_url, presence: true
+  validates :position, presence: true
+  validates :subsection_id, presence: true
+
+  validates :subsection_id, uniqueness: true
```

Comment:
the validation for subsection uniqueness isn't what we want - we want to be able to add multiple slides to a subsection